### ISSUE TITLE
Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise

### ISSUE THREAD

**Issue by bamsir** [archfz/cypress-terminal-report#144]
```
CypressError: Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise.

The command that returned the promise was:

  > `cy.wait()`

The cy command you invoked inside the promise was:

  > `cy.wait()`

Because Cypress commands are already promise-like, you don't need to wrap them or return your own promise.
```

We have some code in a test that looks like
```
    return cy
      .wait(200)
      .get(`ul li .nav-item`)
      .contains('title', { matchCase: false })
      .should('be.visible')
      .click({ waitForAnimations: true })
      .wait(100);
```

All tests using this code execute just fine until we load `installLogsCollector()`. Once this is loaded all the waits (and the click with animation wait) start throwing the cypress error above. Sometimes the command invoked inside the promise is `cy.task` which we do not use anywhere in our codebase.

Currently using
```
"cypress": "9.6.1",
"cypress-terminal-report": "3.5.2",
```

**Comment by archfz** [comment]
Can you confirm whether this issue happens in case the cy wait is not chained on other commands. Try to call it only on cy.

**Comment by bamsir** [comment]
I'll do some more digging and see if I can get a small shareable repro. I posted before that only since I saw a similar issue in the past that didn't require it. I knew it was a long shot but gave it a go. 

The wait is returned simply because we use cypress chaining a good bit, it really isn't a necessity here though. The errors are throw before that line, so I didn't test removing it;

This is in a framework independent Nx monorepo, but the app in question is an angular 13 app.  We still use the Page object pattern, and this code is in a shareable page class for our company's reusable tab component.  Unfortunately there are timeouts inside that shared component's code (and we can't control it), thus the manual waits.  The page object is called directly in the cypress test and uses a wrapper to reduce redundancy. Roughly something like below

I fear there's more impacting this than just that pattern. I'll close this until I get more useful information for you. Thanks for the quick reply.
